### PR TITLE
feat-signup, login and authentication

### PR DESCRIPTION
**Pull request by therajatg**
1. made the login page.
2. gave option to login with valid credentials or as a guest.
3. After login the user will be automatically directed to the page where he actually wanted to go.
4. The login page will be removed from history after successful login so in case user clicks back button after successful login, he won't see the login page.
5. made the signup page.
6. storing the encodedToken after successful signup in useAuth context.
7. on logout making the token null.

### CODE REVIEW

File: src/frontend/components/navbar/Navbar.js [hunk 1]
```diff
@@ -1,9 +1,23 @@
...
+import { Link, useNavigate } from "react-router-dom";
 import { IoMdArrowDropdown } from "react-icons/io";
+import { useAuth } from "../../contexts/authContext";
 
 export function Navbar() {
+  let { authState, authDispatch } = useAuth();
+  let { token } = authState;
+  const navigate = useNavigate();
+  console.log(token);
+
```
Comment (prathmeshjagtap) [inline src/frontend/components/navbar/Navbar.js:12]: you can remove console.log

File: src/frontend/pages/login/Login.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+export function Login() {
+  const { authState, authDispatch } = useAuth();
+  const { user } = authState;
+  const [flag, setFlag] = useState(false);
+  const location = useLocation();
+  const navigate = useNavigate();
+
+  async function loginHandler(e) {
+    console.log(e);
```
Comment (prathmeshjagtap) [inline src/frontend/pages/login/Login.js:15]: you can remove console.log

File: src/frontend/pages/login/Login.js [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+      localStorage.setItem("token", res.data.encodedToken);
+      authDispatch({ type: "TOKEN", payload: res.data.encodedToken });
+      navigate(location.state.from.pathname, { replace: true });
+    } catch (err) {
+      console.log(err);
+    }
+  }
+
+  async function guestHandler(e) {
+    console.log(e);
```
Comment (prathmeshjagtap) [inline src/frontend/pages/login/Login.js:31]: same here

**Review by prathmeshjagtap — APPROVED** [review]
The code is good and Well structured, just remove the console. logs 
It will be better if you can add a preview of the functionality

**Review by parulgupta2001 — APPROVED** [review]
Code looks clean.
 Good to merge

**Pull request merged**